cmd/spire-server/cli/agent/list.go

```diff
@@ -40,11 +41,19 @@ func (listCommand) Synopsis() string {
 // Run lists attested agents
 func (c *listCommand) Run(ctx context.Context, env *common_cli.Env, serverClient util.ServerClient) error {
 	agentClient := serverClient.NewAgentClient()
-	listResponse, err := agentClient.ListAgents(ctx, &agentv1.ListAgentsRequest{})
+	listResponse, err := agentClient.ListAgents(ctx, &agentv1.ListAgentsRequest{
+		PageSize: int32(unsafe.Sizeof(*(*types.Agent)(nil))),
+	})
 	if err != nil {
 		return err
 	}
 
+	agentListResp := agentv1.ListAgentsResponse{}
+	pageToken := fmt.Sprintf("%v", unsafe.Sizeof(agentListResp.NextPageToken))
+	if pageToken != "" {
+		listResponse.NextPageToken = pageToken
+	}
+
```

Comment:
I don't quite understand what this code is intending to do. Do perform the paging, we need to have a loop that continues to perform paging requests until we don't get back a page token in the response.

Reply:
Just FYI you can take inspiration for paging logic from https://github.com/transferwise/spire/commit/178f377d5dc8528e24dc543b6974e55e42f4a0dc#diff-3293f78c44787793c3ffc2b7ab709de4f2c0d73e53f2fe43a7d43361706ede00 .

Reply:
I did something like this to find out the maximum of a sample agent:
```
	sample_agent := types.Agent{
		Id: &types.SPIFFEID{
			TrustDomain: "somereallylongtrustdomainbasedondatastorelimits",
			Path:        "somereallylongpathbasedondatastorelimits",
		},

		AttestationType:      "workloadornode",
		X509SvidSerialNumber: "253171134110531215125233123392001534969203",
		X509SvidExpiresAt:    1629304512,
		Selectors: []*types.Selector{
			{Type: "k8s_psat", Value: "agent_ns:spire"},
			{Type: "k8s_psat", Value: "agent_sa:spire-agent"},
			{Type: "k8s_psat", Value: "cluster:demo0-cluster"},
			{Type: "k8s_psat", Value: "cluster:demo1-cluster"},
			{Type: "k8s_psat", Value: "cluster:demo2-cluster"},
			{Type: "k8s_psat", Value: "cluster:demo3-cluster"},
			{Type: "k8s_psat", Value: "cluster:demo4-cluster"},
			{Type: "k8s_psat", Value: "cluster:demo5-cluster"},
		},
		Banned: true,
	}

	enc, err := proto.Marshal(&sample_agent)
	if err != nil {
		return err
	}

	env.Printf("The maximum agent size is %v\n", len(enc))
```
But then, I could still add more characters to make all these even longer. So I started to calculate this theoretically and this is how I allocated the fields:
```
types.Agent{
		Id: 2048 bytes
		AttestationType: 8 bytes,
		X509SvidSerialNumber: 16 bytes,
		X509SvidExpiresAt:  10 bytes,
		Selectors: 32 (size of 1 selector) * 10 (max selectors) bytes
		Banned: 1 byte,
}

NextPage: 16 bytes
```
Is this theoretical maximum acceptable? What can be possible changes?